### PR TITLE
fix: Drop `is_between()` date series method

### LINKED ISSUE #1188: Drop `is_between()` DateSeries method
The `is_between()` method is the exclusive complement to the inclusive `is_on_or_between()` and was added (I think) just for the sake of consistency because the other inclusive methods (`is_on_or_before()` and `is_on_or_after()`) have exclusive complements.

But as @wjchulme has pointed out, the potential for confusion here is quite high as it would be very easy for a casual reader to assume that it was inclusive. And on top of that, an exclusive date range function doesn't seem all that useful.

Slack thread here:
https://bennettoxford.slack.com/archives/C04DVD1UQC9/p1680707203503939

### PR DESCRIPTION
**Pull request by milanwiedemann**
Drop `is_between()` as described in #1188. For acceptance test studies in this repo suggest an identical alternative using `is_on_or_between()` together with adding and subtracting one day to/from the interval. 

Closes #1188

### CODE REVIEW

File: tests/acceptance/external_studies/openprompt-long-covid-vaccines/analysis/dataset_definition_longcovid_prevaccine.py [hunk 1]
```diff
@@ -115,16 +115,16 @@
 # covid tests ------------------------------------------------------------
 dataset.first_test_post_lc = sgss_covid_all_tests \
     .where(sgss_covid_all_tests.is_positive) \
-    .where(sgss_covid_all_tests.specimen_taken_date.is_between(dataset.pt_start_date, dataset.pt_end_date)) \
```
Comment (evansd) [inline tests/acceptance/external_studies/openprompt-long-covid-vaccines/analysis/dataset_definition_longcovid_prevaccine.py:118]: This is a correct translation of the semantics of the original query, but I wonder if it's what the author actually intended? I think we'll need to confirm with them in each case.

**Review by evansd — COMMENTED** [review]
As mentioned in [this comment](https://github.com/opensafely-core/ehrql/issues/1188#issuecomment-1618101179) I think we'll need to check the intention with the study authors of the few uses of `is_between()` that we want to remove.

File: tests/acceptance/external_studies/ons-mental-health/analysis/dataset_definition_ons_cis_new.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 # range
 #######################################################################################
 
-ons_cis = schema.ons_cis.where(schema.ons_cis.visit_date.is_on_or_between(start_date, end_date))
+ons_cis = schema.ons_cis.where(schema.ons_cis.visit_date.is_on_or_between(start_date + days(1), end_date - days(1)))
```
[the remark this answers is not in the document]
Comment (milanwiedemann) [reply tests/acceptance/external_studies/ons-mental-health/analysis/dataset_definition_ons_cis_new.py:22]: ahhh it doesnt! thanks for spotting

File: tests/acceptance/external_studies/openprompt-long-covid-vaccines/analysis/dataset_definition_longcovid_prevaccine.py (hunk 1 above)
Comment (milanwiedemann) [reply tests/acceptance/external_studies/openprompt-long-covid-vaccines/analysis/dataset_definition_longcovid_prevaccine.py:118]: I checked with the study author, they did intend to use the `is_between()` as it was implemented, i.e., excluding the start and end date

**Comment by milanwiedemann** [comment]
we decided to keep this functionality because it was the intended behaviour in a research study. but we will rename it to `is_between_but_not_on()` to avoid confusion and follow the naming convention of similar methods, work continues here #1405

**Pull request closed without merging**